### PR TITLE
MAINT: Drop versioneer for hatch backend, fully embrace pyproject.toml

### PR DESCRIPTION
**Pull request by mgxd**
Following groundwork laid out in nipreps/fmriprep#2914

### TODO
- [x] Move wrapper to pyproject.toml
- [x] Merge https://github.com/nipreps/niworkflows/pull/769 and cut release / pin to dev master

### CODE REVIEW

**Comment by mgxd** [comment]
@effigies since you implemented in fmriprep, mind giving this a quick look?

File: .gitignore [hunk 1]
```diff
@@ -98,3 +98,6 @@ ENV/
 work/
 testdata/
 .vscode/
+
+nibabies/_version.py
+wrapper/**/_version.py
```
Comment (effigies) [inline .gitignore:103]: Simpler and less brittle if you reorganize:

```suggestion
_version.py
```

File: wrapper/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+[tool.hatch.build.targets.wheel]
+packages = ["src/nibabies_wrapper"]
+
+[tool.hatch.version]
+source = "vcs"
+raw-options = { version_scheme = "nipreps-calver", root = ".." }
+
+[tool.hatch.build.hooks.vcs]
+version-file = "src/nibabies_wrapper/_version.py"
+
```
Comment (effigies) [inline wrapper/pyproject.toml:52]: Do you want the `bdist_wheel` section here?

```suggestion

[tool.distutils.bdist_wheel]
universal = true

```

File: pyproject.toml [hunk 3]
```diff
@@ -1,9 +1,124 @@
...
+
+[tool.hatch.version]
+source = "vcs"
+raw-options = { version_scheme = "nipreps-calver" }
+
+[tool.hatch.build.hooks.vcs]
+version-file = "nibabies/_version.py"
+
+[tool.distutils.bdist_wheel]
+universal = true
```
Comment (effigies) [inline pyproject.toml:98]: We don't support Python 2 except in the wrapper. I don't think you want this here.

```suggestion
```

File: wrapper/src/nibabies_wrapper/__main__.py [hunk 4]
```diff
@@ -735,4 +738,6 @@ def main():
 
 
 if __name__ == "__main__":
+    if '__main__.py' in sys.argv[0]:
+        sys.argv[0] = '%s -m %s' % (sys.executable, __name__)
```
Comment (effigies) [inline wrapper/src/nibabies_wrapper/__main__.py:742]: We know `__name__ == "__main__"`. We want the `__name__` of the parent module.

```suggestion
    if '__main__.py' in sys.argv[0]:
        from . import __name__ as module
        sys.argv[0] = '%s -m %s' % (sys.executable, module)
```

**Review by effigies — APPROVED** [review]
Overall LGTM. Have not actually looked to see if checkmarks are green...

File: pyproject.toml (hunk 3 above)
Comment (mgxd) [reply pyproject.toml:98]: I was following https://github.com/nipreps/fmriprep/commit/c4acb74ba77c5e7f6eea6649d31e485c9cb84162

Comment (effigies) [reply pyproject.toml:98]: That looks like an error in my PR.

**Pull request merged**